bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -101,25 +102,45 @@ final class BagsManager(executor: ScheduledThreadPoolExecutor) {
       synchronized {
         val key = obs.getNodeKey
         state += key
-        executor.schedule(new Thread {
-          setPriority(Thread.NORM_PRIORITY - 1)
 
-          override def run(): Unit =
+        def bagsStatus(status: BagsStatus) = bagsStatuses.synchronized {
+          val oldState = bagsStatuses.get(key)
+          bagsStatuses += ((key, status))
+          notifyBagsStatusListeners(observation, oldState, Some(status))
+        }
+        def bagsClearStatus() = bagsStatuses.synchronized {
+          val oldState = bagsStatuses.get(key)
+          bagsStatuses -= key
+          notifyBagsStatusListeners(observation, oldState, None)
+        }
+
```

Comment:
Are you sure you want to synchronize on the immutable value referenced by a `var`? I have a hard time understanding the intent. Why isn't `bagsStatuses` part of `BagsState`? We now have two independent hunks of shared state.